### DISCUSSION TITLE
Is there any way to list files in the current directory with double clicking TAB?

### DISCUSSION THREAD

**Discussion by csy19960309** [daniele77/cli#156] [Q&A] [answered]
I have checked usage of auto-completion, but still stay confused how to customize it. 
For example, how can I specify what will be displayed when I invoke auto-completion? I know names of menus are listed, so what about anything else? Like current directory files?
I used to achieve this feature with GNU readline. Now I switch to this cli lib because I think it is better.

**Comment by daniele77** [comment, marked as answer]
Hi @csy19960309 
currently, the library does not provide a way to customize autocompletion.
The autocompletion works only with commands and menus (i.e., just the command part of the line is autocompleted, not the parameters).
However,  we can plan to add a new feature to the library: a new parameter type "file". When you provide a command with that parameter, the library should autocomplete the parameter with files on the current directory (bash style). The same with a parameter "directory".
In some way, this new feature will overlap a bit with issue #69.
I created a new issue anyway:  #157

**Reply by csy19960309** [reply]
> Hi @csy19960309 currently, the library does not provide a way to customize autocompletion. The autocompletion works only with commands and menus (i.e., just the command part of the line is autocompleted, not the parameters). However, we can plan to add a new feature to the library: a new parameter type "file". When you provide a command with that parameter, the library should autocomplete the parameter with files on the current directory (bash style). The same with a parameter "directory". In some way, this new feature will overlap a bit with issue #69. I created a new issue anyway: #157

Alright. Sorry for my late reply. It it nice of you to plan to bring so many incredible features. Should I close this issue?

**Reply by daniele77** [reply]
I moved your issue to discussions (i.e. the forum), so I guess you don't need to close it.
